### PR TITLE
Add Element Call participant limit

### PR DESCRIPTION
**Pull request by SimonBrandner**
Requires https://github.com/vector-im/element-web/pull/23431

<hr>

---
Here's what your changelog entry will look like:

## ✨ Features
 * Add Element Call participant limit ([\#9358](https://github.com/matrix-org/matrix-react-sdk/pull/9358)).

### CODE REVIEW

File: src/components/views/voip/CallView.tsx [hunk 1]
```diff
@@ -197,6 +199,11 @@ export const Lobby: FC<LobbyProps> = ({ room, connect, children }) => {
         }
     }, [connect, setConnecting]);
 
+    const tooltip = useMemo(() => {
+        if (connecting) return _t("Connecting");
+        if (callIsFull) return _t("Sorry - this call is currently full");
```
Comment (robintown) [inline src/components/views/voip/CallView.tsx:204]:
```suggestion
        if (callIsFull) return _t("Sorry—this call is currently full");
```

em dash to make the copy editor in me happy

File: test/components/views/voip/CallView-test.tsx [hunk 2]
```diff
@@ -163,6 +165,26 @@ describe("CallLobby", () => {
...
+        it("disables join button when the participant limit has been exceeded", async () => {
+            const sdkConfigCopy = SdkConfig.get();
+            const bob = mkRoomMember(room.roomId, "@bob:example.org");
+            const carol = mkRoomMember(room.roomId, "@carol:example.org");
+
+            jest.spyOn(SdkConfig, "get").mockImplementation((name: string) => {
+                if (name === "element_call") return { participant_limit: 2 };
+                if (Boolean(name)) return sdkConfigCopy[name];
+                return sdkConfigCopy;
+            });
```
Comment (robintown) [inline test/components/views/voip/CallView-test.tsx:178]: Does a real call to `SdkConfig.put` not work?

**Review by robintown — COMMENTED** [review]
I think it would be nice if we could show the tooltip on the buttons in call tiles as well

File: src/components/views/voip/CallView.tsx [hunk 3]
```diff
@@ -197,6 +199,11 @@ export const Lobby: FC<LobbyProps> = ({ room, connect, children }) => {
...
+    }, [connecting, callIsFull]);
```
Comment (duxovni) [inline src/components/views/voip/CallView.tsx:205]: What's the rationale for `useMemo` here, and on line 342? These don't seem like expensive operations.

Comment (SimonBrandner) [reply src/components/views/voip/CallView.tsx:205]: Is there a reason not to use it?

Comment (robintown) [reply src/components/views/voip/CallView.tsx]: As the [memo all the things](https://attardi.org/why-we-memo-all-the-things/) argument goes, not really. And in the future React is supposed to move to automatic memoization of everything, anyways

**Review by robintown — APPROVED** [review]

**Pull request merged**